File: WoofIOS/SwiftPackages/Data/Repositories/Sources/Repositories/Account/AccountRepository.swift
import Combine
import DI
import DataSource
import Models

@Single
public final class AccountRepository {
    private let dataSource: AccountDataSourceImplementing
    private let profileMapper: DogProfileDTOToDomainMapper

    public func getAccount() -> AnyPublisher<DogProfile, AccountError> {
        dataSource.getAccount()
            .map { [profileMapper] in profileMapper($0) }
            .mapError { _ in AccountError.unavailable }
            .eraseToAnyPublisher()
    }
}

File: WoofAndroid/data/repositories/src/main/kotlin/com/perrystreet/woof/repositories/account/AccountRepository.kt
package com.perrystreet.woof.repositories.account

import com.perrystreet.woof.datasource.account.IAccountDataSource
import com.perrystreet.woof.models.dog.DogProfile
import com.perrystreet.woof.repositories.dogs.mapper.DogProfileDTOToDomainMapper
import io.reactivex.rxjava3.core.Single

@org.koin.core.annotation.Single
class AccountRepository(
    private val dataSource: IAccountDataSource,
    private val profileMapper: DogProfileDTOToDomainMapper,
) {
    fun getAccount(): Single<DogProfile> = dataSource.getAccount().map { profileMapper(it) }
}
